knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -50,6 +58,44 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	resources, err := kourier.Resources(ingressNamespace, mgr.GetClient())
+	if err != nil {
+		return err
+	}
+
+	// Watch for Kourier resources.
+	gvkToKourier := map[schema.GroupVersionKind][]types.NamespacedName{}
+	for _, u := range resources {
+		gvkToKourier[u.GroupVersionKind()] =
+			append(gvkToKourier[u.GroupVersionKind()], types.NamespacedName{Namespace: u.GetNamespace(), Name: u.GetName()})
+	}
+
+	for _, u := range resources {
+		u.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   u.GroupVersionKind().Group,
+			Kind:    u.GroupVersionKind().Kind,
+			Version: u.GroupVersionKind().Version,
+		})
```

Comment:
I still don't quite grasp why this is needed? You set a groupVersionKind that you get from the very same resource here? Could you elaborate?

Reply:
Sorry, it was my misunderstanding. As you said, it is just set the same resource and we don't need this. I removed them.